server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1006,50 +1006,49 @@ private TsFileProcessor getOrCreateTsFileProcessorIntern(long timeRangeId,
 
   private TsFileProcessor createTsFileProcessor(boolean sequence, long timePartitionId)
       throws IOException, DiskSpaceInsufficientException {
-    String baseDir;
-    if (sequence) {
-      baseDir = DirectoryManager.getInstance().getNextFolderForSequenceFile();
-    } else {
-      baseDir = DirectoryManager.getInstance().getNextFolderForUnSequenceFile();
-    }
+    DirectoryManager directoryManager = DirectoryManager.getInstance();
+    String baseDir = sequence ? directoryManager.getNextFolderForSequenceFile()
+        : directoryManager.getNextFolderForUnSequenceFile();
+
     fsFactory.getFile(baseDir, storageGroupName).mkdirs();
 
     String filePath =
         baseDir + File.separator + storageGroupName + File.separator + timePartitionId
             + File.separator
             + getNewTsFileName(timePartitionId);
 
-    TsFileProcessor tsFileProcessor;
     VersionController versionController = getVersionControllerByTimePartitionId(timePartitionId);
+    TsFileProcessor tsFileProcessor = getTsFileProcessor(sequence, filePath, versionController);
+    tsFileProcessor.addCloseFileListeners(customCloseFileListeners);
+    tsFileProcessor.addFlushListeners(customFlushListeners);
+
+    tsFileProcessor.setTimeRangeId(timePartitionId);
+    return tsFileProcessor;
+  }
+
+  private TsFileProcessor getTsFileProcessor(boolean sequence, String filePath,
```

Comment:
`newTsFileProcess()` is better.
And,  I think the following should also be in the method.
```
    tsFileProcessor.addCloseFileListeners(customCloseFileListeners);
    tsFileProcessor.addFlushListeners(customFlushListeners);

    tsFileProcessor.setTimeRangeId(timePartitionId);
```

Reply:
Good idea, i will follow your suggestion